refactor(install-trivalent.sh): Move from dnf4 to dnf5 preserving GPG-checking behaviour

srprca opened the pull request:
Previously, DNF4 was used instead of DNF5 for downloading the Trivalent package, as DNF4 performs GPG check after download and DNF5 does not. Neither DNF4 nor DNF5 check the signature of a package installed from outside of a repository (by local path or URL) by default. This commit switches to using DNF5 to download the package (without checking the signature at that point), and subsequently forcing a signature check when installing this package from a local file.

srprca committed:
refactor(install-trivalent.sh): Move from dnf4 to dnf5 preserving GPG-checking behaviour

Previously, DNF4 was used instead of DNF5 for downloading the Trivalent package, as DNF4 performs GPG check after download and DNF5 does not. Neither DNF4 nor DNF5 check the signature of a package installed from outside of a repository (by local path or URL) by default. This commit switches to using DNF5 to download the package (without checking the signature at that point), and subsequently forcing a signature check when installing this package from a local file.

srprca commented:
The only side effect of this commit I can think of is that previously if the downloaded file was bad, the script terminated on line 24, presumably with the corrupt/compromised file removed by DNF4 itself (can't check this), and now it will terminate later, with the file remaining in place.

RoyalOughtness commented on files/scripts/install-trivalent.sh:
```diff
@@ -43,6 +44,7 @@ wget "https://github.com/secureblue/Trivalent/releases/download/${trivalent_vers
 
-dnf install "${trivalent_rpm}" -y
+# Forcing GPG check of a package installed from outside a repository
+dnf5 --setopt=localpkg_gpgcheck=True install "${trivalent_rpm}" -y
```
how does dnf5 know which repo to check against?

srprca replied:
Ah, indeed! It checks against all keys in the RPM database, and since I already had secureblue's key there, it worked for me. It doesn't work after I remove this key, and we are exactly in this setting. I have now updated the commit to explicitly retrieve the GPG key path from the `.repo` file and import it to the RPM database. The check can now proceed as expected.

RoyalOughtness replied:
That's not exactly what I meant. Was the build breaking before? If so, then this is necessary yes. If it wasn't breaking, then this isn't necessary.

srprca replied:
Okay, so your comment made me realize that I didn't in fact follow the established best practices with this PR: I just checked this specific script on my "regular" Fedora system, but not as a part of the secureblue image build process. So I guess let's put this PR on hold for a while, until I set up a proper secureblue development environment and confirm what the minimal necessary changes are.

RoyalOughtness replied:
@srprca you can just swap the cosign keys out on your branch and then run the build in github actions

srprca committed:
refactor(install-trivalent.sh): dnf4 -> dnf5 preserving GPG-checking

Previously, DNF4 was used instead of DNF5 for downloading the Trivalent package, as DNF4 performs GPG check after download and DNF5 does not. Neither DNF4 nor DNF5 check the signature of a package installed from outside of a repository (by local path or URL) by default. This commit switches to using DNF5 to download the package (without checking the signature at that point), and subsequently forcing a signature check when installing this package from a local file.

srprca committed:
Merge branch 'secureblue:live' into patch-1

srprca commented:
Will create a cleaner version of this PR today.
